sqlglot/expressions.py

```diff
@@ -2298,10 +2329,6 @@ class Identifier(Expression):
     def quoted(self) -> bool:
         return bool(self.args.get("quoted"))
 
-    @property
-    def hashable_args(self) -> t.Any:
-        return (self.this, self.quoted)
```

Comment:
Seems like there's a bug in main:

```python
>>> import sqlglot
>>> sqlglot.parse_one("create table #x", "tsql").this.this != sqlglot.exp.to_identifier("x")
False
```

I think it's worth adding a test that asserts this inequality.